File: leetcode/python/two-sum.py
from typing import List
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        value_index = {} #mapping of value to its index
        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in value_index:
                return [i, value_index[complement]]
            else:
                value_index[nums[i]] = i
        return []

File: leetcode/java/TwoSum.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> indexOf = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (indexOf.containsKey(complement)) {
                return new int[] { indexOf.get(complement), i };
            }
            indexOf.put(nums[i], i);
        }
        return new int[] {};
    }
    public int[] Naive(int[] nums, int target){
        int n = nums.length;
        for(int i = 0 ; i < n; i++){
            for(int j = i+1; j < n; j++){
                if(nums[i]+nums[j] == target){
                    return new int[] {i,j};
                }
            }
        }
        return new int[]{};
    }
}
